### DISCUSSION TITLE
JUnit 5 rerunFailingTestsCount

### DISCUSSION THREAD

**Discussion by robertheath** [reportportal/reportportal#1562] [Q&A] [answered]
When I use the rerunFailingTestsCount option in the Maven Surefire plugin to automatically rerun test failures they show up in ReportPortal as separate tests... i.e. the first run of a test fails, gets reported to ReportPortal, the test runs again and passes, test is reported to ReportPortal as a separate test rather than a rerun of the same test. Is there a way to configure ReportPortal to consider the second test run a rerun or would this be a feature request?

**Comment by HardNorth** [comment, marked as answer]
According to official documentation `maven-surefire-plugin` supports JUnit 4 only.

Nevertheless, even though it is supported by the plugin there is no ability to find out from a test listener whether it's rerun or not. Hence there is no option to support such feature.

**Reply by pankajneog** [reply]
Hi @robertheath, did you got any solution for the retry?? I'm also facing the same for cucumber JUnit framework

**Comment by pulse00** [comment]
@HardNorth would it be possible to manually handle the rerun of single tests by using extension points of the ReportPortalExtension?

Even without any surefire logic involved, is it possible to make reportportal recognize a test as a rerun if it has been rerun inside a single launch?
